test/src/unittests/rhythm/test_rhythmextractor2013.py

```diff
@@ -201,23 +197,19 @@ def testImpulseTrainDegara(self):
         # Check bpm intervals
         expectedBpmIntervals = [60/90., 60/140., 60/200.]
         self._assertVectorWithinVector(result[4], expectedBpmIntervals)
-
         result = self._runInstance(self._createPulseTrainCombo(),method="degara")
         expectedBpmVector = [50, 100, 200]
-
-
         # bpm: here rhythmextractor is choosing 0.5*expected_bpm, that's why we are
         # comparing the resulting bpm with the expected_bpm_vector:
         self._assertVectorWithinVector([result[0]], expectedBpmVector, 1.)
         
-        #TODO decide on tolerance
-        self._assertVectorWithinVector(result[1], expectedTicks, 0.03)
+        #TODO decide on menaingful tolerances
```

Comment:
One way to approach this decision is to plot the estimated and expected tick positions. This can help us to see if the test is failing because of a a few particular ticks with a high error margin, or if the estimation is wrong for all ticks

Reply:
If I set the tick Margen to 0.25 I can get the test to pass. I found by trial and error that the lowest margin is 2.303..something

Nevertheless if I do a print to compare expected vs actual values I see a large difference
(This can be seen looking at the data  of "Expected" vs "actual" ticks, even without the need for a plot)
So I don't know why this function call works
self._assertVectorWithinVector(result[1], expectedTicks, 0.24)

Degara with expectedBpmIntervals = [60/90., 60/140., 60/200.] and expectedBpmVector = [50, 100, 200]
Expected Ticks:

[0.1, 0.5285714285714286, 0.9571428571428572, 1.3857142857142857, 1.8142857142857143, 2.242857142857143, 2.6714285714285713, 3.1, 3.5285714285714285, 3.9571428571428573, 4.385714285714286, 4.814285714285714, 5.242857142857143, 5.671428571428572, 6.1, 6.5285714285714285, 6.957142857142857, 7.385714285714286, 7.814285714285714, 8.242857142857142, 8.67142857142857, 9.1, 9.528571428571428, 9.957142857142857]
Actual Ticks
[ 0.6501587  1.3235373  1.9853061  2.6470747  3.3204534  3.993832
  4.655601   5.3173695  5.9907484  6.652517   7.3142858  7.987664
  8.661043   9.322812   9.98458   10.657959  11.319728  11.981497
 12.654875  13.328254  13.990023  14.651792  15.32517   15.9985485
 16.671926  17.345306  17.832924  18.250883  18.668842  19.156462
 19.620861  20.08526   20.51483   20.944399  21.373968  21.803537
 22.233107  22.662676  23.092245  23.521814  23.939774  24.369343
 24.798912  25.22848   25.658049  26.087618  26.517187  26.946756
 27.376326  27.794285  28.223854  28.653423  29.094603  29.535782
 30.18594   30.78966   31.393377  31.997097  32.589207  33.181316
 33.785034  34.388752  34.99247   35.59619   36.188297  36.780407
 37.384125  37.987846  38.591564  39.183674  39.77578   40.30984
 40.89034   41.494057  42.09778   42.689884  43.281994  43.88571
 44.489433  45.09315   45.69687   46.28898   46.88109   47.484806
 48.088524  48.692245  49.295963  49.888073  50.48018   51.0839
 51.68762   52.291336  52.895054  53.487164  54.079273  54.68299
 55.286713  55.89043   56.49415   57.097866  57.689976  58.282085
 58.885803  59.48952   60.093243  60.69696   61.28907   61.88118
 62.484898  63.088615  63.692333  64.29605   64.89977  ]

Reply:
This values look completely off. But also the time scale is very different: the actual ticks get up to 64 seconds. The expected ticks are 10 seconds max. Why is that?